spec/controllers/callbacks_controller_spec.rb

```diff
@@ -2,95 +2,91 @@
 
 describe CallbacksController, :omniauth do
   describe '#create' do
-    context 'with existing user' do
-      let(:user) { create :user }
-      before do
-        request.env['omniauth.auth'] = mock_auth(user)
-        request.env['devise.mapping'] = Devise.mappings[:user]
+    context 'existing user' do
+      context 'with valid token' do
+        let(:user) { create :user }
+        before do
+          request.env['omniauth.auth'] = mock_auth(user)
+          request.env['devise.mapping'] = Devise.mappings[:user]
+        end
+
+        it "doesn't create a new User object" do
+          expect { post :github }.to_not change{ User.count }
+        end
+
+        it 'creates a session' do
+          expect(session).to be_empty
+          post :github
+          expect(session).not_to be_empty
+        end
       end
 
-      it "doesn't create a new User object" do
-        expect { post :github }.to_not change{ User.count }
-      end
+      context 'with valid token and not in whitelisted organisation' do
```

Comment:
I don't get it — what's different here, in comparison with `context 'existing user'`?
How do you make sure that the user set up in this context won't be a member of whitelisted orgs?